src/main/java/com/webapp/TextBook/validation/SharedValidation/loginUserInfoValidation/LoginUserInfoValidatorImpl.java

```diff
@@ -0,0 +1,135 @@
+            }
+
+
+            // commence bi-if logic -- failure of prerequisite condition does not
+            // bar second input (password) from being validated
+            if (!user.get_username().matches(RegexPatternContainer.STUDENT_ID_PATTERN.pattern())) {
+                // add to error binding list
+                errorList.add(new ErrorBinding<String>(LoginUserInfo.NOMINAL_USERNAME, "Invalid 919 number. Please follow format (919######).", user.get_username()));
+            }
+
+            // second input validation (password)
+            // tests prefix of password
+            // two variations exist: password w/ or w/o domain
+            if (!user.get_password().matches(RegexPatternContainer.S_NUMBER_PREFIX.pattern())) {
+                errorList.add(new ErrorBinding<String>(LoginUserInfo.NOMINAL_PASSWORD, "Password does not follow S-Number form (S######). Please revise", null));
+
+            }
+            else {
+
+                // captures if their is a suffix attach to password input
+                // applied to see if
+                //     !haveSuffix: suffix (domain) needs to be spliced if present
+                //     haveSuffix: binding error generated if suffix not present
+                // given: prefix handles if injection or nonsensical additions appended
+                final boolean inputHasSuffix = user.get_password().matches(RegexPatternContainer.S_NUMBER_SUFFIX.pattern());
+
+                // wants: suffix at end
+                // got: no suffix
+                if (this.haveSuffix && !inputHasSuffix) {
+                    // add error to binding list
+                    errorList.add(new ErrorBinding<String>(LoginUserInfo.NOMINAL_PASSWORD, "Your S-Number should does not retain a domain. Please revise.", null));
+
+                }
+                // wants: no suffix at end
+                // got: suffix at end
+                else if (!this.haveSuffix && inputHasSuffix)
+                    user.set_password(
+                            user.get_password().replaceFirst("@.*", "")
+                    );
+
+            }
+            if(!errorList.isEmpty()){ //Error list is not empty
+                //if binding error list is not empty then set constraint validator’s message
+                // to the abstract method’s json creator helper.
+                constraintContext.disableDefaultConstraintViolation();
+
+                constraintContext.buildConstraintViolationWithTemplate(ErrorBinding.ErrorBindingJsonHelper.CreateJsonStringFromErrorBindings(errorList));
+
+            }
+        }
+        catch (ErrorBindingException e){
+            System.out.println("Error binding failed\n" + e.getStackTrace());
+            constraintContext.buildConstraintViolationWithTemplate(SharedValidationState.GENERIC_JSON_ERROR_MESSAGE);
+        }
+        catch(Exception exception){
+            // for when conversion of binding list fails upon
+            // error event json generation
```

Comment:
internal error print
revise message to follow form.

Internal errors follow this form
"Internal Error Occured in  ( (nameOfClass)--(nameOfMethod) ): (errorDescHere)"

Reply:
should return false too

Reply:
https://github.com/NicholasPierce1/TextBookServices/commit/a6dc03adf1b8434acb0c5c1bc2f76ab78d2ad803